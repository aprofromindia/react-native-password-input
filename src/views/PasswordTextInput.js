// @flow

import * as React from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';
import colors from '../configs/AppConfig';

type Props = { placeholder: string };

type State = { text: string, hideText: boolean };

export default class PasswordTextInput extends React.PureComponent<
  Props,
  State
> {
  state = {
    text: '',
    hideText: true,
  };

  onPress = () => this.setState({ hideText: !this.state.hideText });

  onChangeText = text => this.setState({ text });

  onShownBackground = (): ?{ backgroundColor: string, borderColor: string } => {
    if (!this.state.hideText) {
      return {
        backgroundColor: colors.lightRed,
        borderColor: colors.borderRed,
      };
    }
    return null;
  };

  onShownText = (): ?{ color: string } => {
    if (!this.state.hideText) {
      return { color: colors.fontRed };
    }
    return null;
  };

  render(): React.Element<*> {
    return (
      <View style={[styles.container, this.onShownBackground()]}>
        <TextInput
          autoCorrect={false}
          keyboardType="default"
          returnKeyType="go"
          ellipsizeMode="end"
          value={this.state.text.toUpperCase()}
          secureTextEntry={this.state.hideText}
          style={styles.input}
          placeholder={this.props.placeholder}
          numberOfLines={1}
          onChangeText={this.onChangeText}
        />
        <TouchableOpacity style={styles.button} onPress={this.onPress}>
          <Text style={this.onShownText()}>
            {this.state.hideText ? 'SHOW' : 'HIDE'}
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: colors.borderColor,
    borderRadius: 2,
    height: 40,
    margin: 16,
    paddingLeft: 10,
  },
  input: { flex: 8 },
  button: { flex: 2, justifyContent: 'center', alignItems: 'center' },
});
