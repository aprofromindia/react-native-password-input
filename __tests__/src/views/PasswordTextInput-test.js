import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import PasswordTextInput from '../../../src/views/PasswordTextInput';

describe('PasswordTextInput Snapshots', () => {
  const wrapper = shallow(<PasswordTextInput placeholder="PASSWORD" />);

  it('should be empty', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should store password', () => {
    wrapper.find('TextInput').simulate('changeText', 'password123');
    expect(wrapper).toMatchSnapshot();
  });

  it('should show password when clicked', () => {
    wrapper.find('TouchableOpacity').simulate('press');
    expect(wrapper).toMatchSnapshot();
  });

  it('should hide password when clicked again', () => {
    wrapper.find('TouchableOpacity').simulate('press');
    expect(wrapper).toMatchSnapshot();
  });
});
