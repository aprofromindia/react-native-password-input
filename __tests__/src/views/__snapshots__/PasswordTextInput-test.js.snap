// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextInput should be empty 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b3b3b3",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "margin": 16,
        "paddingLeft": 10,
      },
      null,
    ]
  }
>
  <TextInput
    autoCorrect={false}
    ellipsizeMode="end"
    keyboardType="default"
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="PASSWORD"
    returnKeyType="go"
    secureTextEntry={true}
    style={
      Object {
        "flex": 8,
      }
    }
    value=""
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={null}
    >
      SHOW
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`PasswordTextInput should hide password when clicked again 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b3b3b3",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "margin": 16,
        "paddingLeft": 10,
      },
      null,
    ]
  }
>
  <TextInput
    autoCorrect={false}
    ellipsizeMode="end"
    keyboardType="default"
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="PASSWORD"
    returnKeyType="go"
    secureTextEntry={true}
    style={
      Object {
        "flex": 8,
      }
    }
    value="PASSWORD123"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={null}
    >
      SHOW
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`PasswordTextInput should show password when clicked 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b3b3b3",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "margin": 16,
        "paddingLeft": 10,
      },
      Object {
        "backgroundColor": "#fff3f3",
        "borderColor": "#cc7d7d",
      },
    ]
  }
>
  <TextInput
    autoCorrect={false}
    ellipsizeMode="end"
    keyboardType="default"
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="PASSWORD"
    returnKeyType="go"
    secureTextEntry={false}
    style={
      Object {
        "flex": 8,
      }
    }
    value="PASSWORD123"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#a41818",
        }
      }
    >
      HIDE
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`PasswordTextInput should store password 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#b3b3b3",
        "borderRadius": 2,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "margin": 16,
        "paddingLeft": 10,
      },
      null,
    ]
  }
>
  <TextInput
    autoCorrect={false}
    ellipsizeMode="end"
    keyboardType="default"
    numberOfLines={1}
    onChangeText={[Function]}
    placeholder="PASSWORD"
    returnKeyType="go"
    secureTextEntry={true}
    style={
      Object {
        "flex": 8,
      }
    }
    value="PASSWORD123"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={null}
    >
      SHOW
    </Text>
  </TouchableOpacity>
</View>
`;
